dump/redis_match.py

```diff
@@ -0,0 +1,278 @@
+        if template['error']:
+            return self.__return_error(template)
+        
+        src = None
+        d_src = ""
+        if req.db:
+            d_src = req.db
+            src = RedisSource()
+        else:
+            d_src = req.file
+            src = JsonSource()
+        
+        if not src.connect(d_src, req.ns):
+            template['error']  = error_dict["CONN_ERR"]
+            return self.__return_error(template)
+        verbose_print("MatchRequest Checks Passed")
+        all_matched_keys = src.getKeys(req.db, req.table, req.key_pattern)
+        if not all_matched_keys or len(all_matched_keys) == 0:
+            template['error'] = error_dict["INV_PTTRN"]
+            return self.__return_error(template)
+        
+        filtered_keys = self.__filter_out_keys(src, req, all_matched_keys)
+        verbose_print("Filtered Keys:" + str(filtered_keys))
+        return self.__fill(src, req, filtered_keys)
+    
+    def __ret_template(self):
+        return {"error" : "", "keys" : [], "return_values" : {}}
+    
+    
+    def __return_error(self, template):
+        verbose_print("MatchEngine: \n" + template['error'])
+        return template
+    
+    def __validate_request(self, req):
+        
+        if not isinstance(req, MatchRequest):
+            return error_dict["INV_REQ"]
+            
+        if not(req.db) and not(req.file):
+            return error_dict["NO_SRC"]
+        
+        if req.db and req.file:
+            return error_dict["SRC_VAGUE"]
+        
+        if not req.db and os.path.exists(req.file):
+            try:
+                with open(req.file) as f:
+                    json.load(f)
+            except ValueError as e:
+                return error_dict["INV_JSON"] 
+        elif not req.db:
+            return error_dict["NO_FILE"]
+        
+        if not(req.file) and req.db not in SonicDBConfig.getDbList():
+            return error_dict["INV_DB"]
+        
+        if not req.table:
+            return error_dict["NO_TABLE"]
+        
+        if not req.key_pattern:
+            return error_dict["NO_KEY"]
+        
+        if not isinstance(req.return_fields, list):
+            return error_dict["BAD_FORMAT_RE_FIELDS"]
+        
+        if not req.just_keys and len(req.return_fields) > 0:
+            return error_dict["JUST_KEYS_COMPAT"]
+        
+        if req.field and not req.value:
+            return error_dict["NO_VALUE"]
+        
+        return ""
+    
+    def __filter_out_keys(self, src, req, all_matched_keys):
+        if not (req.field):
+            return all_matched_keys
+        
+        filtered_keys = []
+        for key in all_matched_keys:
+            f_values = src.hget(req.db, key, req.field)
+            if "," in f_values: # Fields Containing Multile Values
+                f_value = f_values.split(",")
+            else:
+                f_value = [f_values]
+            if req.value in f_value:
```

Comment:
I don't think you should split values and consider a match. For use cases that require so please have it as an option provided through request. If that option is not specified, please match complete list.